### ISSUE TITLE
Time index gets duplicated due to order of aggregation operations

### ISSUE THREAD

**Issue by andersy005** [NCAR/intake-esm#169]
As it's been pointed out in https://github.com/NCAR/intake-esm-datastore/issues/37, https://github.com/NCAR/intake-esm/issues/160; for some queries, intake-esm returns datasets with incorrect time axis. After some investigations, it appears that this issue is mainly due to the order of operations. I tried two configurations and got two different results:

Order of Operation 1:

```json
   "aggregations": [
      {
        "type": "union",
        "attribute_name": "variable_id"
      },
        {
        "type": "join_new",
        "attribute_name": "member_id",
        "options": { "coords": "minimal", "compat": "override" }
      },
      {
        "type": "join_new",
        "attribute_name": "dcpp_init_year",
        "options": { "coords": "minimal", "compat": "override" }
      },
      {
        "type": "join_existing",
        "attribute_name": "time_range",
        "options": { "dim": "time", "coords": "minimal", "compat": "override" }
      }

    ]
```

In `intake-esm`, these aggregations translate into 

`xr.concat(dsets, dim="time")` --> `xr.concat(dsets, dim="dcpp_init_year")` --> `xr.concat(dsets, dim="member_id")` --> `xr.merge(dsets)`

**Note:** *The above configuration yields datasets with **correct** time axis.*

----

Order of Operation 2:
```
 "aggregations": [
      {
        "type": "union",
        "attribute_name": "variable_id"
      },
      {
        "type": "join_existing",
        "attribute_name": "time_range",
        "options": { "dim": "time", "coords": "minimal", "compat": "override" }
      },
      {
        "type": "join_new",
        "attribute_name": "member_id",
        "options": { "coords": "minimal", "compat": "override" }
      },
      {
        "type": "join_new",
        "attribute_name": "dcpp_init_year",
        "options": { "coords": "minimal", "compat": "override" }
      }
    ]
```

In `intake-esm`, these aggregations translate into 

`xr.concat(dsets, dim="dcpp_init_year")` --> `xr.concat(dsets, dim="member_id")` --> `xr.concat(dsets, dim="time")` --> `xr.merge(dsets)`

**Note:** *The above configuration yields datasets with **incorrect (duplicated)** time axis.*

Notebook with details: https://gist.github.com/andersy005/04d7416cd92bf857e48caf88fcec8d4b#file-xarray-merge-tool-ipynb

**Comment by andersy005** [comment]
From the above findings, it appears that the `join_existing` step should always been done before any of the `join_new` steps

Cc @matt-long, @rabernat

**Comment by andersy005** [comment]
Closing this issue for now (resolved in #171)
